IntegrationTests/Services/AWSS3IntegrationTests/S3FlexibleChecksumsTests.swift

```diff
@@ -0,0 +1,95 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import XCTest
+import AWSS3
+@testable import ClientRuntime
+
+final class S3FlexibleChecksumsTests: S3XCTestCase {
+
+    private func testPutGetObject(
+        withChecksumAlgorithm algorithm: S3ClientTypes.ChecksumAlgorithm,
+        objectNameSuffix: String, upload: ByteStream
+        ) async throws {
+        let objectName = "flexible-checksums-s3-test-\(objectNameSuffix)"
+
+        let input = PutObjectInput(
+            body: upload,
+            bucket: bucketName,
+            checksumAlgorithm: algorithm,
+            key: objectName
+        )
+
+        let response = try await client.putObject(input: input)
+        let unwrappedResponse = try XCTUnwrap(response)
+
+        // Verify the checksum response based on the algorithm used.
+        XCTAssertNotNil(try XCTUnwrap(getChecksumResponse(from: unwrappedResponse, with: algorithm)))
+
+        let getInput = GetObjectInput(
+            bucket: bucketName,
+            checksumMode: S3ClientTypes.ChecksumMode.enabled,
+            key: objectName
+            )
+        let responseGet = try await client.getObject(
+            input: getInput
+            ) // will error for normal payloads if checksum mismatch
+        XCTAssertNotNil(responseGet.body) // Ensure there's a body in the response.
```

Comment:
In addition to checking it's not nil, I think it's important to ensure the returned object really is the one that was just uploaded. Could we add a check for it? Something like `XCTAssertEqual(<response>, upload)` where `<response>` is the corresponding counterpart in `responseGet`.

Reply:
They have to be equal if we don't get a checksum mismatch exception! Since ChecksumMode = `ENABLED` we will calculate checksum of the response body and compare it to the one sent to us by S3. If the data is the same then checksum will be the same and we'll be good. If not then we won't. But actually, Ill add a check to make sure validation occurred.

Reply:
Oh I see , that makes sense 👍